capstone-backend/src/test/java/com/google/sps/servlets/SearchServletTest.java

```diff
@@ -113,7 +113,8 @@ public boolean checkPopulatedVolumeDataObject(VolumeData book) {
   public void ensureJsonResponseIsInCorrectFormat() {
     // Response should be an array of JSON objects, in the form of:
     // [{book1Info}, {book2Info}...]
-
+    // Remove all whitespace as we only care about the beginning and ending characters
+    jsonResponse = jsonResponse.replaceAll("\\s","");
```

Comment:
Is this actually helpful in the tests if you're only looking at the beginning and ending characters? It's extra code and extra processing so it should really only be included if it has a clear benefit.

Reply:
This test was originally intended to confirm that the response from the servlet was a proper JSON object. With the change to prettyFormatting (which greatly improves the human readability of the response), the characters were no longer exactly where I expected them to be with the new spacing changes.

Given that the rest of the tests (which are parsing JSON) work based on the JSONResponse being correctly formatted JSON, I think it is reasonable to remove this test since the extra work of the new replaceAll makes the test no longer worth it. I have removed it.